### DISCUSSION TITLE
Is fromNode ( field extensions ) public ?

### DISCUSSION THREAD

**Discussion by tonyhallett** [gatsbyjs/gatsby#33913] [Help]
`@proxy` has [tests ](https://github.com/gatsbyjs/gatsby/blob/79aee20caf199961174a1ca385907a4be3360181/packages/gatsby/src/schema/extensions/__tests__/field-extensions.js#L1331) for it.

The following through extensions, not directives.

`@dateformat` [has support](https://github.com/gatsbyjs/gatsby/blob/79aee20caf199961174a1ca385907a4be3360181/packages/gatsby/src/schema/types/date.ts#L283)

`@link' [has support](https://github.com/gatsbyjs/gatsby/blob/79aee20caf199961174a1ca385907a4be3360181/packages/gatsby/src/schema/resolvers.ts#L342) 

`@fileByRelativePath` [has support](https://github.com/gatsbyjs/gatsby/blob/79aee20caf199961174a1ca385907a4be3360181/packages/gatsby/src/schema/resolvers.ts#L475)

--
The [default field resolver](https://github.com/gatsbyjs/gatsby/blob/79aee20caf199961174a1ca385907a4be3360181/packages/gatsby/src/schema/resolvers.ts#L614) that they all pass through to.

**Comment by LekoArts** [comment]
Yes, it is a public API. We don't really document it though.

**Comment by tonyhallett** [comment]
@LekoArts thanks for your response
Should dateformat, link, fileByRelativePath directives be changed to support it ?
